<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>peco</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">peco-paper</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jhsiao999/peco-paper">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">peco</h1>

</div>

<div id="TOC">
<ul>
<li><a href="#overview">Overview</a></li>
<li><a href="#the-software-peco">The software peco</a></li>
<li><a href="#the-analysis">The analysis</a></li>
<li><a href="#citation">Citation</a></li>
<li><a href="#downloading-the-data-files">Downloading the data files</a></li>
<li><a href="#other-information">Other information</a></li>
</ul>
</div>

<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-10-03
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>peco-paper/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190814code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190814)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190814code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190814)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjdblischakpecopapertreecdab18c04e69147709bd8ea49c77f0101a1328c6targetblankcdab18ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jdblischak/peco-paper/tree/cdab18c04e69147709bd8ea49c77f0101a1328c6" target="_blank">cdab18c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjdblischakpecopapertreecdab18c04e69147709bd8ea49c77f0101a1328c6targetblankcdab18ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/jdblischak/peco-paper/tree/cdab18c04e69147709bd8ea49c77f0101a1328c6" target="_blank">cdab18c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
working directory clean
</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML (<code>docs/index.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jdblischak/peco-paper/blob/cdab18c04e69147709bd8ea49c77f0101a1328c6/analysis/index.Rmd" target="_blank">cdab18c</a>
</td>
<td>
John Blischak
</td>
<td>
2020-10-03
</td>
<td>
Update citation.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jdblischak/peco-paper/95c39abec06caee694238f49aa8d1a8a4a665bbe/docs/index.html" target="_blank">95c39ab</a>
</td>
<td>
jhsiao999
</td>
<td>
2020-01-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jdblischak/peco-paper/blob/ea63cb6a73c3f6edfa344f2e1287ab8c89c615eb/analysis/index.Rmd" target="_blank">ea63cb6</a>
</td>
<td>
jhsiao999
</td>
<td>
2020-01-26
</td>
<td>
homepage
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jdblischak/peco-paper/blob/206dbf5479978a755cd3df50aa46b90532601ee1/analysis/index.Rmd" target="_blank">206dbf5</a>
</td>
<td>
jhsiao999
</td>
<td>
2020-01-12
</td>
<td>
updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jdblischak/peco-paper/35b41c28c878450a7933a18cf426edcec1b69c0a/docs/index.html" target="_blank">35b41c2</a>
</td>
<td>
jhsiao999
</td>
<td>
2019-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jdblischak/peco-paper/blob/944d881d17827b515629826ba9be78260976fec1/analysis/index.Rmd" target="_blank">944d881</a>
</td>
<td>
jhsiao999
</td>
<td>
2019-09-16
</td>
<td>
fix code/ hyperlink
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jdblischak/peco-paper/bf95f6861e0bb58527fab9ffb85e91507f63c51e/docs/index.html" target="_blank">bf95f68</a>
</td>
<td>
jhsiao999
</td>
<td>
2019-09-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jdblischak/peco-paper/blob/33b9a3431175376a7eef81ea5da931fca7479d65/analysis/index.Rmd" target="_blank">33b9a34</a>
</td>
<td>
jhsiao999
</td>
<td>
2019-09-13
</td>
<td>
updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jdblischak/peco-paper/2d3a9907e34115539f7643919d578f3bd7077533/docs/index.html" target="_blank">2d3a990</a>
</td>
<td>
jhsiao999
</td>
<td>
2019-09-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jdblischak/peco-paper/blob/60e328130a38ca592f2502f79d555be450b2b085/analysis/index.Rmd" target="_blank">60e3281</a>
</td>
<td>
jhsiao999
</td>
<td>
2019-09-13
</td>
<td>
wflow_publish(c(&quot;analysis/index.Rmd&quot;, &quot;analysis/access_data.Rmd&quot;, &quot;analysis/license.Rmd&quot;,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jdblischak/peco-paper/60a309fea1fdb81e921a63dfae621b00c3795fb6/docs/index.html" target="_blank">60a309f</a>
</td>
<td>
jhsiao999
</td>
<td>
2019-09-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jdblischak/peco-paper/blob/37ffff6291e62bf0bd8101a3c90358b5c8647d6a/analysis/index.Rmd" target="_blank">37ffff6</a>
</td>
<td>
jhsiao999
</td>
<td>
2019-09-06
</td>
<td>
update table of content
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jdblischak/peco-paper/ba2d64757ccd31e8480ccf899bb8e58b1fcfbf35/docs/index.html" target="_blank">ba2d647</a>
</td>
<td>
Joyce Hsiao
</td>
<td>
2019-08-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jdblischak/peco-paper/blob/63517ffa7a0d1414c7e98d316843cc90f6db2e7c/analysis/index.Rmd" target="_blank">63517ff</a>
</td>
<td>
Joyce Hsiao
</td>
<td>
2019-08-14
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="overview" class="section level2">
<h2>Overview</h2>
<p><strong>peco</strong> is a supervised approach for predicting <em>continuous</em> cell cycle phase using single-cell RNA-seq (scRNA-seq) data. The approach is described in <a href="https://doi.org/10.1101/gr.247759.118">our paper</a>.</p>
<p>We use this site to document and share the code used to produce our analysis. Please feel free to explore. Comments and feedbacks are welcome!</p>
</div>
<div id="the-software-peco" class="section level2">
<h2>The software peco</h2>
<p>We quantified continous cell cycle phase using FUCCI fluorescence imaging and trained peco to predict this continuous cell cycle phase using scRNA-seq data collected from six human cell lines. <a href="https://doi.org/10.1101/gr.247759.118">Our paper</a> showed that peco produces robust cell cycle phase predictions using strong cyclic genes - genes which expression levels oscillate along the cell cycle.</p>
<p>The software <strong>peco</strong> will be released in <a href="https://www.bioconductor.org/developers/release-schedule/">Biconductor 3.11</a>. This release will use the latest <a href="https://cran.r-project.org/">R3.6.1</a>.</p>
<p>The development version is available on <a href="https://github.com/jhsiao999/peco">GitHub</a>. To install the development version,</p>
<pre class="r"><code>devtools::install_github(&quot;jhsiao999/peco&quot;)
library(peco)</code></pre>
</div>
<div id="the-analysis" class="section level2">
<h2>The analysis</h2>
<ul>
<li><p><a href="https://jhsiao999.github.io/peco-paper/access_data.html">Process scRNA-seq and imaging data</a></p></li>
<li><p><a href="https://jhsiao999.github.io/peco-paper/npreg_trendfilter_quantile.html">Estimate the cyclic trends of gene expression levels</a></p></li>
<li><p><a href="https://jhsiao999.github.io/peco-paper/eval_on_our_data.html">Compare the performance of peco with other methods on our data</a></p></li>
<li><p><a href="https://jhsiao999.github.io/peco-paper/eval_on_leng2015_data.html">Compare the performance of peco with other methods using Leng et al. 2015 data</a></p></li>
<li><p><a href="https://jhsiao999.github.io/peco-paper/predict_thinned_data.html">Assess the performance of peco in thinned data</a></p></li>
<li><p><a href="https://github.com/jhsiao999/peco-paper/tree/master/code">Produce figures shown in our paper</a></p></li>
</ul>
</div>
<div id="citation" class="section level2">
<h2>Citation</h2>
<p><a href="https://doi.org/10.1101/gr.247759.118">Characterizing and inferring quantitative cell-cycle phase in single-cell RNA-seq data analysis</a>.</p>
</div>
<div id="downloading-the-data-files" class="section level2">
<h2>Downloading the data files</h2>
<p>You have two main options for downloading the data files. First, you can manually download the individual files by clicking on the links on this page or navigating to the files in the <a href="https://github.com/jhsiao999/peco-paper">peco-paper</a> GitHub repository. This is the recommended strategy if you only need a few data files.</p>
<p>Second, you can install <a href="https://git-lfs.github.com/">git-lfs</a>. To handle large files, we used Git Large File Storage (LFS). This means that the files that you download with <code>git clone</code> are only plain text files that contain identifiers for the files saved on GitHub's servers. If you want to download all of the data files at once, you can do this with after you install git-lfs.</p>
<p>To install git-lfs, follow their instructions to download, install, and setup (<code>git lfs install</code>). Alternatively, if you use conda, you can install git-lfs with <code>conda install -c conda-forge git-lfs</code>. Once installed, you can download the latest version of the data files with <code>git lfs pull</code>.</p>
</div>
<div id="other-information" class="section level2">
<h2>Other information</h2>
<ul>
<li><p>GEO record <a href="http://www.ncbi.nlm.nih.gov/%20geo/query/acc.cgi?acc=GSE121265">GSE121265</a> for all raw and processed sequencing data.</p></li>
<li><p>We also provide the processed data in TXT format as a gzip compressed tarball on the <a href="https://giladlab.uchicago.edu/wp-content/uploads/2019/02/Hsiao_et_al_2019.tar.gz">Gilad lab website</a>.</p></li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.3 (2020-02-29)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5       whisker_0.4      knitr_1.30       magrittr_1.5    
 [5] R6_2.4.1         rlang_0.4.7      stringr_1.4.0    tools_3.6.3     
 [9] xfun_0.17        git2r_0.27.1     htmltools_0.5.0  ellipsis_0.3.1  
[13] yaml_2.2.1       digest_0.6.25    rprojroot_1.2    tibble_3.0.3    
[17] lifecycle_0.2.0  crayon_1.3.4     later_1.1.0.1    vctrs_0.3.4     
[21] promises_1.1.1   fs_1.5.0         glue_1.4.2       evaluate_0.14   
[25] rmarkdown_2.3    stringi_1.5.3    compiler_3.6.3   pillar_1.4.6    
[29] backports_1.1.10 httpuv_1.5.4     pkgconfig_2.0.3 </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
