<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Joyce Hsiao" />


<title>Estimate cyclic trend of gene expression levels</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">peco-paper</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jhsiao999/peco-paper">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Estimate cyclic trend of gene expression levels</h1>
<h4 class="author"><em>Joyce Hsiao</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-01-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>peco-paper/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190814code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190814)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190814code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190814)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjhsiao999pecopapertree12d976a3bc9a66e022b1d0ed84721a39d9289333targetblank12d976aa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jhsiao999/peco-paper/tree/12d976a3bc9a66e022b1d0ed84721a39d9289333" target="_blank">12d976a</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjhsiao999pecopapertree12d976a3bc9a66e022b1d0ed84721a39d9289333targetblank12d976aa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  code/note_wo_w_pca.R
    Untracked:  data/data_training_test/
    Untracked:  data/eset-filtered.rds
    Untracked:  data/fit.trend.perm.highmiss.rds
    Untracked:  data/fit.trend.perm.lowmiss.rds
    Untracked:  data/fit_diff_cyclone.rds
    Untracked:  data/fit_diff_oscope.rds
    Untracked:  data/fit_diff_peco.rds
    Untracked:  data/fit_diff_recat.rds
    Untracked:  data/fit_diff_seurat.rds
    Untracked:  data/intensity.rds
    Untracked:  data/leng2015_data.rds
    Untracked:  data/leng_fucci_oscope_29genes.rda
    Untracked:  data/leng_fucci_recat.rda
    Untracked:  data/leng_geneinfo.txt
    Untracked:  data/log2cpm.quant.rds
    Untracked:  data/macosko-2015.rds
    Untracked:  data/nmeth.3549-S2.xlsx
    Untracked:  data/ourdata_cyclone_NA18511.rds
    Untracked:  data/ourdata_cyclone_NA18855.rds
    Untracked:  data/ourdata_cyclone_NA18870.rds
    Untracked:  data/ourdata_cyclone_NA19098.rds
    Untracked:  data/ourdata_cyclone_NA19101.rds
    Untracked:  data/ourdata_cyclone_NA19160.rds
    Untracked:  data/ourdata_oscope_366genes.rda
    Untracked:  data/ourdata_peco_NA18511_top005genes.rds
    Untracked:  data/ourdata_peco_NA18855_top005genes.rds
    Untracked:  data/ourdata_peco_NA18870_top005genes.rds
    Untracked:  data/ourdata_peco_NA19098_top005genes.rds
    Untracked:  data/ourdata_peco_NA19101_top005genes.rds
    Untracked:  data/ourdata_peco_NA19160_top005genes.rds
    Untracked:  data/ourdata_phase_cyclone.rds
    Untracked:  data/ourdata_phase_seurat.rds
    Untracked:  data/ourdata_recat.rda
    Untracked:  data/sce-filtered.rds

Unstaged changes:
    Modified:   analysis/index.Rmd
    Modified:   code/run_seurat.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhsiao999/peco-paper/blob/12d976a3bc9a66e022b1d0ed84721a39d9289333/analysis/npreg_trendfilter_quantile.Rmd" target="_blank">12d976a</a>
</td>
<td>
jhsiao999
</td>
<td>
2020-01-26
</td>
<td>
estimate cyclic trends of gene expression levels
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>We used trendfilter to estimate cyclic trend of gene expression levels for each gene. We computed proportion of variance explained (PVE) by the cyclic trend for each gene and quantify the signficance of cycliic trend by permutation-based p-values.</p>
</div>
<div id="set-up" class="section level2">
<h2>Set-up</h2>
<p>Load packages</p>
<pre class="r"><code>library(SingleCellExperiment)
library(dplyr)
library(matrixStats)
library(edgeR)
library(circular)
library(peco)</code></pre>
<p>Load data</p>
<pre class="r"><code>sce &lt;- readRDS(&quot;data/sce-final.rds&quot;)
sce &lt;- sce[grep(&quot;ENSG&quot;, rownames(sce)),]
pdata &lt;- data.frame(colData(sce))
fdata &lt;- data.frame(rowData(sce))

sce &lt;- data_transform_quantile(sce)
log2cpm_quantNormed &lt;- assay(sce, &quot;cpm_quantNormed&quot;)
log2cpm_beforequant &lt;- assay(sce, &quot;cpm&quot;)

# derive and rotate cell cycle phase
pca &lt;- prcomp(cbind(pdata$rfp.median.log10sum.adjust,
                    pdata$gfp.median.log10sum.adjust))
theta &lt;- coord2rad(pca$x)
theta_final &lt;- shift_origin(as.numeric(theta), 3*pi/4)

log2cpm_quantNormed &lt;- log2cpm_quantNormed[,order(theta_final)]
log2cpm_beforequant &lt;- log2cpm_beforequant[,order(theta_final)]

plot(log2cpm_quantNormed[&quot;ENSG00000170312&quot;,],
     main = &quot;CDK1&quot;, ylab = &quot;Normalized gene expression values&quot;, 
     xlab = &quot;FUCCI phase&quot;)</code></pre>
<p><img src="figure/npreg_trendfilter_quantile.Rmd/load-data-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="evaluate-data-after-standardizing-expression" class="section level2">
<h2>Evaluate data after standardizing expression</h2>
<p>Map log2cpm expression to standard normal distribution. The transformation is non-linear. Sort N expression values from the largest to the smalles. General N standard normal random variable. For the non-zero expression value, find the correspnoding standard normal random variable that has the same quantile and subsitute the value with the corresponding normal random variable value. We then find the stanadard normal random variable values correspond to non-zero expression values and randomly assign these non-zero expression values to a standard normal random variable value.</p>
<ul>
<li><p>For genes with relatively low fraction of deteted cells, this method allows to move the zero expression values closer to the non-zero expression value.</p></li>
<li><p>For genes with high fraction of undetected cells, this method creates a</p></li>
</ul>
<p>Check genes with low/high fraction of undetected cells.</p>
<pre class="r"><code>ii.high &lt;- order(rowMeans(log2cpm_beforequant &gt; 0), decreasing = F)[1:5]
par(mfcol=c(3,5))
for (i in 1:5) {
  plot(log2cpm_beforequant[ii.high[i],], ylab = &quot;log2 CPM expression values&quot;)
  plot(log2cpm_quantNormed[ii.high[i],], ylab = &quot;quantile-normalized log2CPM&quot;)
  plot(x=log2cpm_beforequant[ii.high[i],], y=log2cpm_quantNormed[ii.high[i],],
       ylab = &quot;quantile-normalized log2CPM&quot;,
       xlab = &quot;log2 CPM expression values&quot;)
}</code></pre>
<p><img src="figure/npreg_trendfilter_quantile.Rmd/check-high-low-undetected-cells-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ii.low &lt;- order(rowMeans(log2cpm_beforequant &gt; 0), decreasing = T)[1:5]
par(mfcol=c(3,5))
for (i in 1:5) {
  plot(log2cpm_beforequant[ii.low[i],], ylab = &quot;log2 CPM expression values&quot;)
  plot(log2cpm_quantNormed[ii.low[i],], ylab = &quot;quantile-normalized log2CPM&quot;)
  plot(x=log2cpm_beforequant[ii.low[i],], y=log2cpm_quantNormed[ii.low[i],],
       ylab = &quot;quantile-normalized log2CPM&quot;,
       xlab = &quot;log2 CPM expression values&quot;)
}</code></pre>
<p><img src="figure/npreg_trendfilter_quantile.Rmd/check-high-low-undetected-cells-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Check genes that we previously found to have cyclical patterns in <a href="https://www.ncbi.nlm.nih.gov/pubmed/12058064">Whitfield et al 2002</a>.</p>
<p>The Whitfeld et al. 2002 list was downloaded from Macosko et al. 2015 (10.1016/j.cell.2015.05.002). Link to the file is <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4481139/bin/NIHMS687993-supplement-supp_data_2.xlsx" class="uri">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4481139/bin/NIHMS687993-supplement-supp_data_2.xlsx</a>.</p>
<pre class="r"><code>macosko &lt;- readRDS(&quot;data/macosko-2015.rds&quot;)

log2cpm_quantNormed_macosko &lt;- log2cpm_quantNormed[rownames(log2cpm_quantNormed) %in%macosko$ensembl,]
macosko_present &lt;- macosko[macosko$ensembl %in% rownames(log2cpm_quantNormed),]

par(mfrow=c(8,5), mar = c(2,2,2,1))
for (g in seq_along(macosko_present$ensembl[1:25])) {
  ind &lt;- rownames(log2cpm_quantNormed_macosko) == macosko_present$ensembl[g]
  plot(log2cpm_quantNormed_macosko[ind,], ylab = &quot;Quantile-normalized log2CPM expression values&quot;,
       main = paste(macosko_present$hgnc[g], &quot;,&quot;, macosko_present$phase[g]),
       pch = 16, cex=.6, ylim = c(-3,3))
}</code></pre>
<p><img src="figure/npreg_trendfilter_quantile.Rmd/check-cyclic-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="fit-trendfilter-to-the-data-after-quantile-normalization" class="section level2">
<h2>Fit trendfilter to the data after quantile normalization</h2>
<pre class="r"><code>fit.trend &lt;- mclapply(1:10, function(g) {
  fit_trendfilter_generic(log2cpm_quantNormed[g,],
                              polyorder = 2)
}, mc.cores=25)
names(fit.trend) &lt;- rownames(log2cpm_quantNormed)

saveRDS(fit.trend, &quot;data/fit.quant.rds&quot;))</code></pre>
<pre class="r"><code># load pre-computed resutls
fit.quant &lt;- readRDS(&quot;data/fit.quant.rds&quot;)

pve &lt;- sapply(fit.quant, &quot;[[&quot;, &quot;trend.pve&quot;)
summary(pve)</code></pre>
<pre><code>      Min.    1st Qu.     Median       Mean    3rd Qu.       Max. 
-0.0043839  0.0000798  0.0002166  0.0008961  0.0004973  0.3266288 </code></pre>
<p>Plot top 10 genes in PVE.</p>
<pre class="r"><code>pve.genes &lt;- names(pve)[order(c(pve), decreasing = T)[1:10]]
par(mfrow=c(2,5))
for (g in 1:length(pve.genes)) {
  ii.g &lt;- which(names(fit.quant)==pve.genes[g])
  plot(log2cpm_quantNormed[rownames(log2cpm_quantNormed)==pve.genes[g],],
       main = fdata[pve.genes[g],]$name, ylab = &quot;Normalized expression&quot;)
  points(fit.quant[[ii.g]]$trend.yy, pch=16, col = &quot;blue&quot;, cex=.7)
}</code></pre>
<p><img src="figure/npreg_trendfilter_quantile.Rmd/pve-top10-genes-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>quickily check top 100 enrichment for cell cycle genes.</p>
<pre class="r"><code>enrich.order &lt;- function(cutoffs, metrics, cyclegenes, allgenes) {
  #  out &lt;- order(mad.ratio$smash.mad.ratio)
  # cutoffs &lt;- c(100, 200, 300)
  cycle.rich &lt;- sapply(cutoffs, function(x) {
    which_top &lt;- order(metrics, decreasing = T)[1:x]
    sig.cycle &lt;- sum(allgenes[which_top] %in% cyclegenes)/x
    non.cycle &lt;- sum(allgenes[-which_top] %in% cyclegenes)/(length(allgenes)-x)
    cbind(as.numeric(sum(allgenes[which_top] %in% cyclegenes)), 
          sig.cycle/non.cycle)
  })
  colnames(cycle.rich) &lt;- cutoffs
  rownames(cycle.rich) &lt;- c(&quot;nsig.genes.cycle&quot;, &quot;fold.sig.vs.nonsig.cycle&quot;)
  cycle.rich
}

macosko &lt;- readRDS(&quot;data/macosko-2015.rds&quot;)
enrich.order(cutoffs = c(100, 200, 300), 
             metrics = pve, cyclegenes = macosko$ensembl,
             allgenes = rownames(log2cpm_quantNormed))</code></pre>
<pre><code>                              100       200   300
nsig.genes.cycle         54.00000 73.000000 86.00
fold.sig.vs.nonsig.cycle 12.78701  8.931377  7.16</code></pre>
</div>
<div id="compute-permutation-based-p-values" class="section level2">
<h2>Compute permutation-based p-values</h2>
<p>Consider two genes, one with large fraction of undetected cells and one with small fraction of undeteted cells. See if the null distribution is similar.</p>
<pre class="r"><code>nperm &lt;- 1000

# choose a gene with high fraction of missing and permute data
set.seed(17)
ii.frac.miss.high &lt;- names(sample(which(rowMeans(log2cpm_beforequant==0) &gt; .8),1))

fit.trend.highmiss &lt;- mclapply(1:nperm, function(g) {
  fit.trendfilter.generic(sample(log2cpm_quantNormed[ii.frac.miss.high,]),
                              polyorder = 2)
}, mc.cores=25)
saveRDS(fit.trend.highmiss, &quot;data/fit.trend.perm.highmiss.rds&quot;)


# choose a gene with low fraction of missing and permute data
set.seed(31)
ii.frac.miss.low &lt;- names(sample(which(rowMeans(log2cpm_beforequant==0) &lt; .1),1))

fit.trend.lowmiss &lt;- mclapply(1:nperm, function(g) {
  fit.trendfilter.generic(sample(log2cpm_quantNormed[ii.frac.miss.low,]),
                  polyorder = 2)
}, mc.cores=25)
saveRDS(fit.trend.lowmiss, &quot;data/fit.trend.perm.lowmiss.rds&quot;)</code></pre>
<p>Turns out the p-value based on permuted data is not the same for genes with low and high fraction of undetected cells.</p>
<pre class="r"><code># load pre-computed results
perm.lowmiss &lt;- readRDS(&quot;data/fit.trend.perm.lowmiss.rds&quot;)

perm.highmiss &lt;- readRDS(&quot;data/fit.trend.perm.highmiss.rds&quot;)

pve.perm.lowmiss &lt;- sapply(perm.lowmiss, &quot;[[&quot;, &quot;trend.pve&quot;)
pve.perm.highmiss &lt;- sapply(perm.highmiss, &quot;[[&quot;, &quot;trend.pve&quot;)

summary(pve.perm.lowmiss)</code></pre>
<pre><code>     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
3.700e-08 4.998e-05 1.388e-04 2.821e-04 3.382e-04 1.617e-02 </code></pre>
<pre class="r"><code>summary(pve.perm.highmiss)</code></pre>
<pre><code>     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
1.340e-07 5.089e-05 1.389e-04 2.664e-04 3.403e-04 4.214e-03 </code></pre>
<pre class="r"><code>par(mfrow=c(1,2))
hist(pve.perm.lowmiss, nclass=30,
     main = &quot;Fraction undetected &lt; 10%&quot;, xlab = &quot;p-value&quot;)
hist(pve.perm.highmiss, nclass=30,
     main = &quot;Fraction undetected &gt; 80%&quot;, xlab = &quot;p-value&quot;)</code></pre>
<p><img src="figure/npreg_trendfilter_quantile.Rmd/load-permdist-results-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Compute p-value based on two different distributions. High consistency between the two.</p>
<p>Use permutated distribution based data with low missing value, which turns out to be more conservative.</p>
<pre class="r"><code>B &lt;- length(pve.perm.lowmiss)
pval.perm.low &lt;- sapply(fit.quant, function(x) (1+sum(pve.perm.lowmiss &gt; as.numeric(x$trend.pve)))/(1+B))
pval.perm.high &lt;- sapply(fit.quant, function(x) (1+sum(pve.perm.highmiss &gt; as.numeric(x$trend.pve)))/(1+B))

summary(pval.perm.low)</code></pre>
<pre><code>    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
0.000999 0.163836 0.365634 0.415336 0.648352 1.000000 </code></pre>
<pre class="r"><code>summary(pval.perm.high)</code></pre>
<pre><code>    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
0.000999 0.150849 0.360639 0.411520 0.630370 1.000000 </code></pre>
<pre class="r"><code>plot(x=pval.perm.low, y=pval.perm.high,
     main = &quot;permutation-based p-values&quot;,
     xlab = &quot;Based on data with low zero fractions&quot;, 
     yalb = &quot;Based on data with high zero fractions&quot;)</code></pre>
<p><img src="figure/npreg_trendfilter_quantile.Rmd/get-permute-pvals-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sum(pval.perm.high &lt; .001)</code></pre>
<pre><code>[1] 278</code></pre>
<pre class="r"><code>sum(pval.perm.low &lt; .001)</code></pre>
<pre><code>[1] 101</code></pre>
<p>Cell cycle signals in signficant cyclic genes.</p>
<pre class="r"><code>which.sig &lt;- pval.perm.low &lt; .001
enrich.sigval &lt;- function(cutoffs, metrics, cyclegenes, allgenes) {
  #  out &lt;- order(mad.ratio$smash.mad.ratio)
  # cutoffs &lt;- c(100, 200, 300)
  cycle.rich &lt;- sapply(cutoffs, function(x) {
    #which_top &lt;- order(metrics, decreasing = T)[1:x]
    sig.cycle &lt;- sum(allgenes[metrics &lt; x] %in% cyclegenes)/sum(metrics &lt; x)
    non.cycle &lt;- sum(allgenes[metrics &gt; x] %in% cyclegenes)/sum(metrics &gt; x)
    cbind(sum(metrics &lt; x), as.numeric(sum(allgenes[metrics &lt; x] %in% cyclegenes)), 
          sig.cycle/non.cycle)
  })
  colnames(cycle.rich) &lt;- cutoffs
  rownames(cycle.rich) &lt;- c(&quot;nsig.genes&quot;, &quot;nsig.genes.cycle&quot;, &quot;fold.sig.vs.nonsig.cycle&quot;)
  cycle.rich
}

enrich.sigval(cutoffs = c(.001, .005, .01), metrics=pval.perm.low,
              cyclegenes = macosko$ensembl,
              allgenes = rownames(log2cpm_quantNormed))</code></pre>
<pre><code>                             0.001      0.005       0.01
nsig.genes               101.00000 476.000000 553.000000
nsig.genes.cycle          54.00000  99.000000  99.000000
fold.sig.vs.nonsig.cycle  12.65925   5.268908   4.502205</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] peco_0.99.10                circular_0.4-93            
 [3] edgeR_3.24.0                limma_3.38.3               
 [5] dplyr_0.8.0.1               SingleCellExperiment_1.4.1 
 [7] SummarizedExperiment_1.12.0 DelayedArray_0.8.0         
 [9] BiocParallel_1.16.0         matrixStats_0.55.0         
[11] Biobase_2.42.0              GenomicRanges_1.34.0       
[13] GenomeInfoDb_1.18.1         IRanges_2.16.0             
[15] S4Vectors_0.20.1            BiocGenerics_0.28.0        

loaded via a namespace (and not attached):
 [1] viridis_0.5.1            genlasso_1.4            
 [3] viridisLite_0.3.0        foreach_1.4.4           
 [5] DelayedMatrixStats_1.4.0 assertthat_0.2.1        
 [7] vipor_0.4.5              GenomeInfoDbData_1.2.0  
 [9] yaml_2.2.0               pillar_1.3.1            
[11] backports_1.1.2          lattice_0.20-38         
[13] glue_1.3.0               digest_0.6.20           
[15] promises_1.0.1           XVector_0.22.0          
[17] colorspace_1.3-2         plyr_1.8.4              
[19] htmltools_0.3.6          httpuv_1.4.5            
[21] Matrix_1.2-17            pkgconfig_2.0.3         
[23] zlibbioc_1.28.0          purrr_0.3.2             
[25] mvtnorm_1.0-11           scales_1.0.0            
[27] HDF5Array_1.10.1         whisker_0.3-2           
[29] later_0.7.5              pracma_2.2.9            
[31] git2r_0.26.1             tibble_2.1.1            
[33] ggplot2_3.2.1            conicfit_1.0.4          
[35] lazyeval_0.2.1           magrittr_1.5            
[37] crayon_1.3.4             evaluate_0.12           
[39] fs_1.3.1                 doParallel_1.0.14       
[41] MASS_7.3-51.1            beeswarm_0.2.3          
[43] geigen_2.3               tools_3.5.1             
[45] scater_1.10.1            stringr_1.3.1           
[47] Rhdf5lib_1.4.3           munsell_0.5.0           
[49] locfit_1.5-9.1           compiler_3.5.1          
[51] rlang_0.4.0              rhdf5_2.26.2            
[53] grid_3.5.1               RCurl_1.95-4.11         
[55] iterators_1.0.12         igraph_1.2.2            
[57] bitops_1.0-6             rmarkdown_1.10          
[59] boot_1.3-20              gtable_0.2.0            
[61] codetools_0.2-15         reshape2_1.4.3          
[63] R6_2.4.0                 gridExtra_2.3           
[65] knitr_1.20               workflowr_1.6.0         
[67] rprojroot_1.3-2          ggbeeswarm_0.6.0        
[69] stringi_1.2.4            Rcpp_1.0.3              
[71] tidyselect_0.2.5        </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] peco_0.99.10                circular_0.4-93            
 [3] edgeR_3.24.0                limma_3.38.3               
 [5] dplyr_0.8.0.1               SingleCellExperiment_1.4.1 
 [7] SummarizedExperiment_1.12.0 DelayedArray_0.8.0         
 [9] BiocParallel_1.16.0         matrixStats_0.55.0         
[11] Biobase_2.42.0              GenomicRanges_1.34.0       
[13] GenomeInfoDb_1.18.1         IRanges_2.16.0             
[15] S4Vectors_0.20.1            BiocGenerics_0.28.0        

loaded via a namespace (and not attached):
 [1] viridis_0.5.1            genlasso_1.4            
 [3] viridisLite_0.3.0        foreach_1.4.4           
 [5] DelayedMatrixStats_1.4.0 assertthat_0.2.1        
 [7] vipor_0.4.5              GenomeInfoDbData_1.2.0  
 [9] yaml_2.2.0               pillar_1.3.1            
[11] backports_1.1.2          lattice_0.20-38         
[13] glue_1.3.0               digest_0.6.20           
[15] promises_1.0.1           XVector_0.22.0          
[17] colorspace_1.3-2         plyr_1.8.4              
[19] htmltools_0.3.6          httpuv_1.4.5            
[21] Matrix_1.2-17            pkgconfig_2.0.3         
[23] zlibbioc_1.28.0          purrr_0.3.2             
[25] mvtnorm_1.0-11           scales_1.0.0            
[27] HDF5Array_1.10.1         whisker_0.3-2           
[29] later_0.7.5              pracma_2.2.9            
[31] git2r_0.26.1             tibble_2.1.1            
[33] ggplot2_3.2.1            conicfit_1.0.4          
[35] lazyeval_0.2.1           magrittr_1.5            
[37] crayon_1.3.4             evaluate_0.12           
[39] fs_1.3.1                 doParallel_1.0.14       
[41] MASS_7.3-51.1            beeswarm_0.2.3          
[43] geigen_2.3               tools_3.5.1             
[45] scater_1.10.1            stringr_1.3.1           
[47] Rhdf5lib_1.4.3           munsell_0.5.0           
[49] locfit_1.5-9.1           compiler_3.5.1          
[51] rlang_0.4.0              rhdf5_2.26.2            
[53] grid_3.5.1               RCurl_1.95-4.11         
[55] iterators_1.0.12         igraph_1.2.2            
[57] bitops_1.0-6             rmarkdown_1.10          
[59] boot_1.3-20              gtable_0.2.0            
[61] codetools_0.2-15         reshape2_1.4.3          
[63] R6_2.4.0                 gridExtra_2.3           
[65] knitr_1.20               workflowr_1.6.0         
[67] rprojroot_1.3-2          ggbeeswarm_0.6.0        
[69] stringi_1.2.4            Rcpp_1.0.3              
[71] tidyselect_0.2.5        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
